packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -461,431 +511,744 @@ describe('Testing authentication module', () => {
-          await utils.initializeWithContext(context);
-          expect(() => authentication.notifyFailure()).toThrow(
-            `This call is only allowed in following contexts: ${JSON.stringify(
-              allowedContexts,
-            )}. Current context: "${context}".`,
-          );
+
+        it(`authentication.getUser should successfully get user profile with ${context} context`, async () => {
+          await framelessPostMock.initializeWithContext(context);
+          const promise = authentication.getUser();
+          const message = framelessPostMock.findMessageByFunc('authentication.getUser');
+          expect(message).not.toBeNull();
+          expect(message.id).toBe(1);
+          expect(message.args[0]).toBe(undefined);
+          framelessPostMock.respondToMessage({
+            data: {
+              id: message.id,
+              args: [false, mockResult],
+            },
+          } as DOMMessageEvent);
+          await expect(promise).rejects.toThrow(mockResult);
         });
-      }
+      });
+    });
+
+    describe('Testing authentication.notifySuccess function', () => {
+      const allowedContexts = [FrameContexts.authentication];
+
+      it('authentication.notifySuccess should not allow calls before initialization', () => {
+        expect(() => authentication.notifySuccess()).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (!allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`authentication.notifySuccess should not allow calls from ${context} context`, async () => {
+            await framelessPostMock.initializeWithContext(context);
+            expect(() => authentication.notifySuccess()).toThrow(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing authentication.notifyFailure', () => {
+      const allowedContexts = [FrameContexts.authentication];
+      it('authentication.notifyFailure should not allow calls before initialization', () => {
+        expect(() => authentication.notifyFailure()).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (!allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`authentication.notifyFailure should not allow calls from ${context} context`, async () => {
+            await framelessPostMock.initializeWithContext(context);
+            expect(() => authentication.notifyFailure()).toThrow(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
```

Comment:
Same here - why no successful frameless unit tests for `authentication.notifyFailure`?

Reply:
I think the reason is that authentication.notifyFailure can't be called from a frameless window. But I'm following up.